### PR TITLE
feat: Add config to control default interdependency ranges

### PR DESCRIPTION
**Pull request by tylerbutler**
## Description

Adds a configuration setting to release groups that controls their interdependency range type. This is a companion change to #15158 and similar changes. That change (and related changes) allowed setting explicit range types in the CI pipeline and used that capability to release builds with `~` interdependencies.

On the other hand, this change adds a setting to the fluid-build configuration that is then used as a default in relevant places rather than the `^` default that was there previously. With this change, the `flub release` and related commands will use the configured interdependency range type unless one is passed as an argument.

I introduced a couple of new types to help consolidate disparate string union types. I also took this opportunity to deprecate and replace the poorly named `exactDepType` flag. It's been replaced by the `interdependencyType` flag, a name which we use throughout the code to refer to this concept.

BREAKING CHANGE:

The `--exactDepType` flag in the `bump` command no longer has a default value. It has also been deprecated. It has been replaced by the `--interdependencyType` flag. The deprecated flag will be removed in an upcoming release.

### CODE REVIEW

File: build-tools/packages/build-cli/src/lib/bump.ts [hunk 1]
```diff
@@ -259,11 +266,11 @@ export async function bumpReleaseGroup(
 	let range: string;
 	if (scheme === "internal" || scheme === "internalPrerelease") {
 		range =
-			exactDependencyType === ""
+			interdependencyType === ""
 				? translatedVersion.version
```
Comment (sonalideshpandemsft) [inline build-tools/packages/build-cli/src/lib/bump.ts:270]: what does `interdependencyType === ""` mean?

File: build-tools/packages/build-cli/src/commands/bump.ts [hunk 2]
```diff
@@ -57,10 +59,19 @@ export default class BumpCommand extends BaseCommand<typeof BumpCommand> {
 			exclusive: ["exact"],
 		}),
 		exactDepType: Flags.string({
+			description:
+				'[DEPRECATED - Use interdependencyType instead.] Controls the type of dependency that is used between packages within the release group. Use "" to indicate exact dependencies.',
+			options: [...InterdependencyRangeTypes],
+			deprecated: {
```
Comment (sonalideshpandemsft) [inline build-tools/packages/build-cli/src/commands/bump.ts:65]: we can deprecate flags too 😲

File: build-tools/packages/build-cli/src/lib/bump.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/lib/bump.ts:270]: I added a comment to clarify.

**Comment by tylerbutler** [comment]
This was merged into #15053.

**Pull request closed without merging**